### PR TITLE
fix: fix publication actions

### LINKED ISSUE #705: (SP: 2) Fix publication actions and toast messages
### **Description**
The publication management system in the admin panel includes actions to publish, cancel publication, or save/exit. Previously, there were inconsistencies and bugs in how these actions mapped to content statuses (e.g., `BaseContentStatuses.Draft` vs `BaseContentStatuses.Published`) and their corresponding success toast messages between the creation (`CreatePublicationsView`) and editing (`EditPublicationsPage`) screens. For instance, the "Publish and exit" action on the edit page was incorrectly saving content as a draft rather than publishing it, and the "Cancel publication" action was displaying generic draft saved toasts.

### Expected result
In the admin panel editor, when administrators perform actions on publications, the editor state must accurately reflect the published state on the client-facing website. Ensuring that actions like "Publicate and exit" transition the status to `Published` rather than `Draft` ensures that the client-facing website immediately displays the updated layouts and content, resolving any discrepancies between the admin panel's reported success actions and the publication's actual visibility to the end user.

### PR DESCRIPTION
**Pull request by telare**
### Description

This PR resolves issue by refactoring and unifying the publication action logic (publishing, saving & exiting, and canceling publication) across the creation and editing workflows. 

Previously, some actions (like `PUBLICATE_AND_EXIT` in the edit page) had mismatched statuses and toast notifications, saving publications as `Draft` instead of `Published`. By centralizing these action mappings in publications.ts under a new configuration object `MENU_ACTION_CONFIGS`, we ensure consistency and correctness in how content status updates and success notifications are handled.

#### 1. Configuration & constants
- **publications.ts**:
  - removed unused `MenuActionId.SAVE_DRAFT` and obsolete toast labels from `CONTENT_MUTATION_RESULTS`.
  - Introduced `MENU_ACTION_CONFIGS` mapping each relevant `MenuActionId` to its designated `BaseContentStatuses` and localized `toastMessage`.
  - marked `PAGE_TITLES` and `ADMIN_TITLE_LABELS` objects `as const` for better type safety.

#### 2. Publications creation, editing, & SEO views
- **(logged_in/publications/[type]/[id]/edit/page.tsx)**:
  - refactored `handleMenuAction` to leverage `MENU_ACTION_CONFIGS`.
  - fixed a bug where `PUBLICATE_AND_EXIT` incorrectly saved the publication as `BaseContentStatuses.Draft` with `draftSaved` toast, changing it to save as `BaseContentStatuses.Published` with `publicationPublished` toast.
- **(logged_in/publications/[type]/create/CreatePublicationsView.tsx)**:
  - updated `handleMenuAction` to fetch action configs from `MENU_ACTION_CONFIGS`, removing local duplication of status/toast definitions.
- **(logged_in/publications/[type]/[id]/seo/page.tsx)**:
  - refactored `handlePublishAndExit`, `handlePublish`, and `handleUnpublish` to use `MENU_ACTION_CONFIGS` for consistent status updates and toast notifications.
  - removed unused `handleSave` and `handleCancel` functions, as direct saving/canceling drafts on the SEO settings view is no longer needed.
- **HeaderRightActions.tsx**:
  - removed unused `onSave` and `onCancel` props from `SeoModeProps` to match the streamlined actions on the SEO page.

#### 3. Test Suites
- **page.test.tsx**:
  - updated test assertions for `PUBLISH` and `PUBLICATE_AND_EXIT` actions to verify the correct status and toast outputs.
- **CreatePublicationsView.test.tsx**:
  - corrected test cases for cancelling a publication to verify the correct success notification message.
- **(logged_in/publications/[type]/[id]/seo/page.test.tsx)**:
  - removed obsolete test assertions checking save/cancel actions from the SEO settings view.
  - updated publication cancellation test to expect the correct success notification message.
  - fixed edge-case test suite for missing `publicationData` to reflect the removal of save/cancel redirects.

Closes #705

## Type of change

- [ ] New feature
- [x] Bug fix
- [x] Refactoring/tech debt
- [ ] Documentation update
- [ ] Other

## How to test

1. Open admin panel
2. Go to the creation page (`/publications/[type]/create`) or edit page (`/publications/[type]/[id]/edit`) of a publication item.
3. Perform the following actions and verify:
   - **Publish**: Check that the publication is successfully updated/created with status `Published` and shows the success toast `"Публікацію опубліковано успішно"`.
   - **Publicate & Exit**: Verify that the publication is saved with status `Published`, displays the correct success toast, and redirects to the publications list.
   - **Cancel publication**: Verify that the publication is saved as `Draft`, shows the success toast `"Публікацію скасовано"`, and redirects.
4. Run tests using `npm run test` or `jest` to ensure all tests pass

## Video / Screenshots
test event: publish and exit upon creation

https://github.com/user-attachments/assets/b8220229-21ab-43d5-989f-43e2a134cea4

test news: publish and exit upon creation

https://github.com/user-attachments/assets/d164ff94-7b03-4ef0-aa1a-061cfbea283f

test media: publish and exit upon creation
note: here you can see an error on the client, this is error with some image urls - this is not related to the current PR

https://github.com/user-attachments/assets/c688207c-bcd0-4837-b1d4-143efbe00338

test news: hiding upon editing

https://github.com/user-attachments/assets/72b8347f-3065-4d67-a7c9-a0badad0e281

test news: hiding upon editing

https://github.com/user-attachments/assets/4d4205d9-f1a3-4465-9a15-f2ff72fd1a13

test media hiding upon editing
https://github.com/user-attachments/assets/8fd1fbc3-b364-452d-9bbf-d2f04976828d

test to see if the post is republished when only the 'Publish' button is clicked
https://github.com/user-attachments/assets/799692ee-7430-4d90-95b5-9b935a8c281a

## Checklist

- [x] Code compiles and runs correctly
- [x] Tests pass successfully
- [X] Linting checks are clean
- [X] No Sonar issues
- [x] Branch is up to date with the target branch
- [x] Linked issue/task included
- [ ] Task moved to the review column on the board

### CODE REVIEW

**Review by DenisGordProgrammer — APPROVED** [review]

File: app/(logged_in)/publications/[type]/[id]/seo/page.tsx [hunk 1]
```diff
@@ -43,19 +42,23 @@ export default function PublicatiosSeoPage() {
   const handleClosePublish = () => setPublishAnchor(null);
 
   const handlePublishAndExit = () => {
-    publicationData?.handleSave(BaseContentStatuses.Published);
-    toast.success(CONTENT_MUTATION_RESULTS.publicationPublished);
+    const { status, toastMessage } = MENU_ACTION_CONFIGS.PUBLICATE_AND_EXIT;
+    publicationData?.handleSave(status);
```
Comment (DenisGordProgrammer) [inline app/(logged_in)/publications/[type]/[id]/seo/page.tsx]: `handlePublishAndExit` fires the toast and redirects synchronously before `handleSave` resolves. PR #695 fixes this exact pattern by making these handlers async and awaiting `handleSave`. Both PRs modify the same lines in this file -- whichever is merged second will need a rebase to keep the await, otherwise the async fix from #695 is lost.

Comment (telare) [reply app/(logged_in)/publications/[type]/[id]/seo/page.tsx]: Good, I will recheck both implementations to make it clear

Comment (telare) [reply app/(logged_in)/publications/[type]/[id]/seo/page.tsx]: Fixed that in the new commit. Currently, I have implemented a correct flow that ensures we await all async code before making `router.push`, and I have also added `toast.error` for better UX if handleSave fails.

**Review by Yushchyk-Roman — APPROVED** [review]
